src/Diagnostics.HealthChecks.AspNetCore/Readme.md

```diff
@@ -9,3 +9,27 @@ The liveness health check is a special type of health check that limits its prob
 - The Health Check implementor will call the `CreateLivenessHttpHealthCheck` method in the [`HealthCheckComposablesExtensions`](./HealthCheckComposablesExtensions.cs) class. This factory method will create a default Http Health Check, but it will tweak the Activity by injecting the `activityMarker` function in the constructor, adding a key to the `Activity` baggage.
 - The Activity will be sent to the endpoint.
 - The endpoint will have to be marked with the [`LivenessCheckActionFilterAttribute`](./LivenessCheckActionFilterAttribute.cs). This filter will check whether there's an `Activity` injected in the request, and whether the `Activity` has the required key in its baggage: if that is the case, the endpoint execution will be short-circuited, returning an HTTP 200 status code, otherwise the method will be executed normally.
+
+### Extension methods
+
+The project offers extension methods to easily register liveness health checks on endpoints that can be specified with one line of configuration. The extension methods are located in the `HealthCheckComposableExtensions`, and they will extend the functionality on the `IHealthCheckBuilder` class available in the DI.
+
+#### AddEndpointHttpHealthCheck
+
+There are two overloads of the `HealthCheckComposableExtensions.AddEndpointHttpHealthCheck`. The first one allows the developer to specify directly the custom `HttpClientParameters` class called `EndpointLivenessHealthCheckParameters`: this class contains all the required information on how to build the HTTP Request to query the endpoint to check.
+
+The second overload has been implemented to offer a similar functionality to the one available using the legacy health checks in the `HealthCheckComposablesExtensions` class: in this overload, all the information previously included in the `EndpointLivenessHealthCheckParameters` class can be defined manually.
+
+For instance, in the following code it is shown how to define an HTTP Health Check on the `/healthz` endpoint using the legacy extension method:
+
+```csharp
+services.AddServiceFabricHealthChecks().AddHttpEndpointCheck("healthz", endpointName, "/healthz");
```

Comment:
The deprecation notice on AddHttpEndpointCheck should be updated to point to the replacement implementation.